docs/concepts/dependencies.md

```diff
@@ -223,6 +223,46 @@ members = [
 ]
 ```
 
+### Platform-specific sources
+
+You can limit a source to a given platform or Python version by providing
+[PEP 508](https://peps.python.org/pep-0508/#environment-markers)-compatible environment markers for
+the source.
+
+For example, to pull `httpx` from GitHub, but only on macOS, use the following:
+
+```toml title="pyproject.toml"
+[project]
+dependencies = [
+  "httpx",
+]
+
+[tool.uv.sources]
+httpx = { git = "https://github.com/encode/httpx", tag = "0.27.2", marker = "sys_platform == 'darwin'" }
+```
+
+By specifying the marker on the source, uv will still include `httpx` on all platforms, but will
+download the source from GitHub on macOS, and fall back to PyPI on all other platforms.
+
+### Multiple sources
+
+You can specify multiple sources for a single dependency by providing a list of sources,
+disambiguated by [PEP 508](https://peps.python.org/pep-0508/#environment-markers)-compatible
+environment markers. For example, to pull in different `httpx` commits on macOS vs. Linux:
+
+```toml title="pyproject.toml"
+[project]
+dependencies = [
+  "httpx",
+]
+
+[tool.uv.sources]
+httpx = [
+  { git = "https://github.com/encode/httpx", tag = "0.27.2", marker = "sys_platform == 'darwin'" },
+  { git = "https://github.com/encode/httpx", tag = "0.24.1", marker = "sys_platform == 'linux'" },
+]
```

Comment:
We definitely can't support the above since we don't support disjoint extras, and those markers are overlapping. I talk about this at length here: https://www.notion.so/astral-sh/PyTorch-10a48797e1ca803ea593c456e9bf23f8?pvs=4#10a48797e1ca80f4ab5af90ebcd77000.

Reply:
I need to see if we already propagate extras here though. I.e., this should probably work, right?

```toml
[project.optional-dependencies]
cu118 = []

[tool.uv.sources]
torch = [
    { index = "torch-cu118", marker = "extra == 'cu118'"},
]

[[tool.uv.index]]
name = "torch-cu118"
url = "https://download.pytorch.org/whl/cu118"
```

Reply:
I do think we _need_ to support it once we support disjoint extras. It makes the mixed CPU/GPU stuff _so_ much better.